src/include/duckdb/main/config.hpp

```diff
@@ -164,6 +165,9 @@ struct DBConfig {
 	//! Replacement open handlers are callbacks that run pre and post database initialization
 	vector<ReplacementOpen> replacement_opens;
 
+	//! Callback to allow users to pass custom create database table functions
+	vector<CreateDatabaseExtension> create_database_extensions;
```

Comment:
Curious: why do we need a vector here? Isn't one enough?

Reply:
Hmm, I mostly just wanted to stay consistent with `vector<ReplacementOpen> replacement_opens;` since that's where the inspiration came from. Also wonder if @Mytherin might have suggestions.